### ISSUE TITLE
GHC 9.4 support

### ISSUE THREAD

**Issue by peterbecich** [freizl/hoauth2#178]
The build appears to fail with GHC 9.4:

```
$  cabal build all --enable-tests                                                                                                                                     1 ↵
Resolving dependencies...
Error: cabal: Could not resolve dependencies:
[__0] trying: hoauth2-2.7.0 (user goal)
[__1] trying: base-4.17.0.0/installed-4.17.0.0 (dependency of hoauth2)
[__2] trying: aeson-2.1.0.0 (dependency of hoauth2)
[__3] trying: hoauth2-demo-1.6.0 (user goal)
[__4] next goal: jose-jwt (dependency of hoauth2-demo)
[__4] rejecting: jose-jwt-0.9.4 (conflict: aeson==2.1.0.0, jose-jwt =>
aeson>=1.5 && <2.1)
[__4] skipping: jose-jwt-0.9.3 (has the same characteristics that caused the
previous version to fail: excludes 'aeson' version 2.1.0.0)
[__4] rejecting: jose-jwt-0.9.2 (conflict: hoauth2-demo => jose-jwt>=0.9.4 &&
<0.10)
[__4] skipping: jose-jwt-0.9.1, jose-jwt-0.9.0, jose-jwt-0.8.0,
jose-jwt-0.7.8, jose-jwt-0.7.7, jose-jwt-0.7.6, jose-jwt-0.7.5,
jose-jwt-0.7.4, jose-jwt-0.7.3, jose-jwt-0.7.2, jose-jwt-0.7.1, jose-jwt-0.7,
jose-jwt-0.6.2, jose-jwt-0.6.1, jose-jwt-0.6, jose-jwt-0.5, jose-jwt-0.4.2,
jose-jwt-0.4.1.1, jose-jwt-0.4.1, jose-jwt-0.4, jose-jwt-0.3.1, jose-jwt-0.3,
jose-jwt-0.2, jose-jwt-0.1 (has the same characteristics that caused the
previous version to fail: excluded by constraint '>=0.9.4 && <0.10' from
'hoauth2-demo')
[__4] fail (backjumping, conflict set: aeson, hoauth2-demo, jose-jwt)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: hoauth2, aeson, jose-jwt,
hoauth2-demo, base
Try running with --minimize-conflict-set to improve the error message.

```

Thank you

**Comment by peterbecich** [comment]
I don't understand
```
rejecting: jose-jwt-0.9.4 (conflict: aeson==2.1.0.0, jose-jwt => aeson>=1.5 && <2.1)
```

`jose-jwt-0.9.4` was revised; it should support `aeson-2.1`: https://hackage.haskell.org/package/jose-jwt-0.9.4/revisions/

**Comment by freizl** [comment]
Works well in `ghc-9.2.5`. I wonder if it's bug in `ghc-9.4.*`. Will look into it.

**Comment by freizl** [comment]
aha, it's because I pin the hackage index here
https://github.com/freizl/hoauth2/blob/main/cabal.project#L2

**Comment by freizl** [comment]
Fixed here and lmk if works for and I'll release a new version. Thanks for reporting the problem.

**Comment by peterbecich** [comment]
Looks good, I have tested it as a dependency of another project:
```
Build profile: -w ghc-9.4.4 -O1
In order, the following will be built (use -v for more details):
 - foo4-0.1.0.0 (test:foo4-test) (first run)
Preprocessing test suite 'foo4-test' for foo4-0.1.0.0..
Building test suite 'foo4-test' for foo4-0.1.0.0..
Running 1 test suites...
Test suite foo4-test: RUNNING...
Test suite foo4-test: PASS
Test suite logged to:
/home/peterbecich/haskell/foo4/dist-newstyle/build/x86_64-linux/ghc-9.4.4/foo4-0.1.0.0/t/foo4-test/test/foo4-0.1.0.0-foo4-test.log
1 of 1 test suites (1 of 1 test cases) passed.
```

```
source-repository-package
  type: git
  location: https://github.com/freizl/hoauth2
  tag: 9ed6aec4341264bcfb1dd9b2b6ed2c9f88ba7bd8
  subdir: hoauth2
```

Thanks!

**Comment by freizl** [comment]
Fixes in https://github.com/freizl/hoauth2/pull/179

**Comment by freizl** [comment]
Also published `2.8.0`
